lambda/authorizer/lambda_functions.py

```diff
@@ -134,3 +143,30 @@ def is_admin(jwt_data: dict[str, Any], admin_group: str, jwt_groups_property: st
         else:
             return False
     return admin_group in current_node
+
+
+def refresh_management_tokens() -> list[str]:
+    """Return secret management tokens if they exist."""
+    global secret_tokens
+
+    if len(secret_tokens) == 0:
+        try:
+            secret_tokens.append(
+                secrets_manager.get_secret_value(
+                    SecretId=os.environ.get("MANAGEMENT_KEY_NAME"), VersionStage="AWSCURRENT"
+                )["SecretString"]
+            )
+            secret_tokens.append(
+                secrets_manager.get_secret_value(
+                    SecretId=os.environ.get("MANAGEMENT_KEY_NAME"), VersionStage="AWSPREVIOUS"
+                )["SecretString"]
+            )
+        except Exception:
+            logger.info(f"No previous secret version for {os.environ.get('MANAGEMENT_KEY_NAME')}")
```

Comment:
can we add the stack trace or actual error in the output too? if we get an exception for something that is not related to getting the previous key then we're outputting a possibly misleading error message.